tests/countries/test_fiji.py

```diff
@@ -0,0 +1,310 @@
+
+    def test_new_years_day(self):
+        name = "New Year's Day"
+
+        self.assertHolidayName(name, (f"{year}-01-01" for year in range(2016, 2050)))
+        dt_obs = (
+            "2017-01-02",
+            "2022-01-03",
+            "2023-01-02",
+            "2028-01-03",
+        )
+        self.assertHolidayName(f"{name} (observed)", dt_obs)
+        self.assertNoNonObservedHoliday(dt_obs)
+
+    def test_good_friday(self):
+        name = "Good Friday"
+
+        dt = (
+            "2018-03-30",
+            "2019-04-19",
+            "2020-04-10",
+            "2021-04-02",
+            "2022-04-15",
+            "2023-04-07",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(2016, 2050))
+
+    def test_easter_saturday(self):
+        name = "Easter Saturday"
+
+        dt = (
+            "2018-03-31",
+            "2019-04-20",
+            "2020-04-11",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(2016, 2050))
+
+    def test_easter_monday(self):
+        name = "Easter Monday"
+
+        dt = (
+            "2018-04-02",
+            "2019-04-22",
+            "2020-04-13",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(2016, 2050))
+
+    def test_girmit_day(self):
+        name = "Girmit Day"
+
+        dt = (
+            "2023-05-15",
+            "2024-05-13",
+            "2025-05-12",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(2023, 2050))
+        self.assertNoHolidayName(name, range(2016, 2023))
+        self.assertNoNonObservedHoliday(dt)
+
+    def test_national_sports_day(self):
+        name = "National Sports Day"
+
+        # Public Holidays
+        dt = (
+            "2016-06-24",
+            "2017-06-30",
+            "2018-06-29",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertNoHolidayName(name, range(2019, 2050))
+
+    def test_constitution_day(self):
+        name = "Constitution Day"
+
+        # Public Holidays
+        self.assertHolidayName(name, (f"{year}-09-07" for year in range(2016, 2023)))
+        self.assertNoHolidayName(name, range(2023, 2050))
+
+        dt_obs = ("2019-09-09",)
+        self.assertHolidayName(f"{name} (observed)", dt_obs)
+        self.assertNoNonObservedHoliday(dt_obs)
+
+        # Workdays.
+        self.assertHolidayName(
+            name, self.workday_holidays, (f"{year}-09-07" for year in range(2023, 2050))
+        )
+        self.assertNoHolidayName(name, self.workday_holidays, range(2016, 2023))
+
+    def test_ratu_sir_lala_sukuna_day(self):
+        name = "Ratu Sir Lala Sukuna Day"
+
+        # Public Holidays.
+        dt = (
+            "2023-05-29",
+            "2024-05-31",
+            "2025-05-30",
+            "2026-05-29",
+            "2027-05-28",
```

Comment:
```suggestion
```
There's no need to get too far in advance, especially since there's no complete certainty that the rule for determining this holiday date is correct.